### PR TITLE
First implementation

### PR DESCRIPTION
**Pull request by maximerety**
Partially based on previous work: https://github.com/TankerHQ/user-token-ruby

Main changes:

* the `rbnacl-libsodium` gem is no longer supported, switched to `rbnacl` + install your own libsodium
* using test vectors from the JS Tanker Identity SDK
* the sinatra app example is closer to a real flow

### CODE REVIEW

**Comment by jeremad** [comment]
I activated the Repo on travis, on next push you'll see CI results

**Review by jeremad — APPROVED** [review]
The test vectors are here, there is a CI... LGTM, but I'm rusty with ruby

File: lib/tanker/identity.rb [hunk 1]
```diff
@@ -1,6 +1,55 @@
...
+
+    def self.user_secret(hashed_user_id)
+      random_bytes = Crypto.random_bytes(USER_SECRET_SIZE - 1)
+      check = Crypto.generichash(random_bytes + hashed_user_id, Crypto::HASH_MIN_SIZE)
+      random_bytes + check[0]
+    end
+
+    public
+
+    def self._deserialize(b64_json)
```
Comment (dmerejkowsky) [inline lib/tanker/identity.rb]: remove this underscore, I don't think it hurts

File: spec/tanker/identity_spec.rb [hunk 2]
```diff
@@ -1,3 +1,63 @@
...
+  end
+
+  describe 'permanent identity' do
+    def assert_user_secret(identity)
+      hashed_user_id = Base64.decode64(@identity['value'])
+      user_secret = Base64.decode64(@identity['user_secret'])
+      expect(hashed_user_id.bytesize).to be 32
+      expect(user_secret.bytesize).to be 32
+
+      controlByte = user_secret.bytes.last
```
Comment (dmerejkowsky) [inline spec/tanker/identity_spec.rb]: javascript dev spotted. use `control_byte`

File: lib/tanker/identity.rb (hunk 1 above)
Comment (maximerety) [reply lib/tanker/identity.rb]: Fixed

File: spec/tanker/identity_spec.rb (hunk 2 above)
Comment (maximerety) [reply spec/tanker/identity_spec.rb]: Fixed

**Review by dmerejkowsky — APPROVED** [review]

**Pull request merged**